distributed/tests/test_client.py

```diff
@@ -5554,27 +5552,20 @@ async def test(s, a, b):
 
             # Can't send complex data
             with pytest.raises(TypeError):
-                future = await c.scatter(inc)
+                future = await c.scatter(SimpleNamespace())
 
-            # can send complex tasks (this uses pickle regardless)
+            # Can send and receive functions tasks (this uses pickle regardless)
             future = c.submit(lambda x: x, inc)
             await wait(future)
-
-            # but can't receive complex results
-            with pytest.raises(TypeError):
-                await c.gather(future, direct=direct)
+            await c.gather(future, direct=direct)
 
             # Run works
             result = await c.run(lambda: 1)
             assert list(result.values()) == [1, 1]
             result = await c.run_on_scheduler(lambda: 1)
             assert result == 1
-
-            # But not with complex return values
-            with pytest.raises(TypeError):
-                await c.run(lambda: inc)
-            with pytest.raises(TypeError):
-                await c.run_on_scheduler(lambda: inc)
+            await c.run(lambda: inc)
+            await c.run_on_scheduler(lambda: inc)
```

Comment:
If the user has specified that they don't want to allow serialization with pickle then these should continue to fail.  Probably we need to feed the list of serializers down wherever serialize is beting called.  I expect that this might be awkward to do when going through msgpack machinery.  Maybe there is some global that we can misuse?